tests/test_scripts.py

```diff
@@ -485,6 +486,70 @@ def test_workspace_digest(tmpdir, script_runner, algorithms, do_json):
         }
 
 
+@pytest.mark.parametrize(
+    "archive",
+    [
+        "https://www.hepdata.net/record/resource/1408476?view=true",
+        "https://doi.org/10.17182/hepdata.89408.v1/r2",
+    ],
+)
+def test_patchset_download(datadir, script_runner, archive):
+    command = f'pyhf contrib download {archive} {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert ret.success
+
+    # Run with all optional flags
+    command = f'pyhf contrib download --verbose --force {archive} {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert ret.success
+
+    command = f'pyhf contrib download --verbose https://www.fail.org/record/resource/1234567 {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert not ret.success
+    assert (
+        "pyhf.exceptions.InvalidArchiveHost: www.fail.org is not an approved archive host"
+        in ret.stderr
+    )
+    command = f'pyhf contrib download --verbose --force https://www.fail.org/record/resource/1234567 {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert not ret.success
+    # Python 3.6 has different return error than 3.7, 3.8
+    assert (
+        "ssl.CertificateError: hostname 'www.fail.org' doesn't match"
+        or "certificate verify failed: Hostname mismatch, certificate is not valid for 'www.fail.org'."
+        in ret.stderr
+    )
+
+
+def test_missing_contrib_extra(datadir, script_runner):
+    module_name = "requests"
+
+    # hide
+    CACHE_MODULE, sys.modules[module_name] = sys.modules[module_name], None
```

Comment:
I guess the only other thing to think about is if this should be made into a decorator.